### PR TITLE
Adding public docstring to Bio.bgzf

### PR DESCRIPTION
**Pull request by Gasta88**
In reference to issue #1203 , I've added public docstring to the module.

This PR is incomplete because I don't understand what the method `seekable()` does at line 710 and 830.

Is this an implementation of the method `seek()`?

### CODE REVIEW

File: Bio/bgzf.py [hunk 1]
```diff
@@ -404,7 +404,7 @@ def BgzfBlocks(handle):
 
 
 def _load_bgzf_block(handle, text_mode=False):
-    """Internal function to load the next BGZF function (PRIVATE)."""
+    """Load the next BGZF function (INTERNAL PRIVATE)."""
```
Comment (peterjc) [inline Bio/bgzf.py]: We've a convention (which may not be documented anywhere?) using ``(PRIVATE)`` at the end of docstrings, and also ``(DEPRECATED)`` and ``(OBSOLETE)`` as per http://biopython.org/wiki/Deprecation_policy but I don't think we've ever used ``INTERNAL`` like this.

How about:

```python
"""Load the next BGZF function (PRIVATE)."""
```

File: Bio/bgzf.py [hunk 2]
```diff
@@ -656,6 +658,7 @@ def read(self, size=-1):
                 return data
 
     def readline(self):
+        """Read single line for the BGZF module."""
```
Comment (peterjc) [inline Bio/bgzf.py]: This isn't for the module, its for the class. How about just:

```python
"""Read a single line from the BGZF file."""
```

File: Bio/bgzf.py [hunk 3]
```diff
@@ -682,6 +685,7 @@ def readline(self):
             return data
 
     def __next__(self):
+        """Return the next value."""
```
Comment (peterjc) [inline Bio/bgzf.py]: Next *line* rather than value (this is the convention for how file iterators work).

File: Bio/bgzf.py [hunk 4]
```diff
@@ -693,9 +697,11 @@ def next(self):
             return self.__next__()
 
     def __iter__(self):
+        """Iterate elements of the BGZF class."""
```
Comment (peterjc) [inline Bio/bgzf.py]: Change to

```python
"""Iterate over the lines in the BGZF file."""
```

File: Bio/bgzf.py [hunk 5]
```diff
@@ -736,6 +748,7 @@ def __init__(self, filename=None, mode="w", fileobj=None, compresslevel=6):
         self.compresslevel = compresslevel
 
     def _write_block(self, block):
+        """Write the datablock into the open file."""
```
Comment (peterjc) [inline Bio/bgzf.py]: Include ``... (PRIVATE).`` at the end please.

**Comment by peterjc** [comment]
Regarding ``.seekable()``, referring to the Python 3 docs https://docs.python.org/2/library/io.html

> Return True if the stream supports random access. If False, seek(), tell() and truncate() will raise IOError.

So how about:

```python
"""Return True indicating the BGZF supports random access."""
```

File: Bio/bgzf.py (hunk 1 above)
Comment (peterjc) [inline Bio/bgzf.py]: Apologies, I missed the word function was being misused here (my own mistake in the original text).

How about:

```python
"""Load the next BGZF block of compressed data (PRIVATE)."""
```

File: Bio/bgzf.py [hunk 6]
```diff
@@ -736,6 +749,7 @@ def __init__(self, filename=None, mode="w", fileobj=None, compresslevel=6):
         self.compresslevel = compresslevel
 
     def _write_block(self, block):
+        """Write the datablock into the open file (PRIVATE)."""
```
Comment (peterjc) [inline Bio/bgzf.py]: To be more explicit, I suggest something like:

```python
"""Write provided data to file as a single BGZF compressed block (PRIVATE)."""
```

**Comment by peterjc** [comment]
I've suggested two more minor tweaks - thanks for looking at this file :)

**Comment by Gasta88** [comment]
Thanks, I've applied the changes.

from now on would you prefer to have a separated PR for each edited file or gather few of them into on PR?

**Comment by peterjc** [comment]
It look like my second comment on the first commit got lost.

For ``_load_bgzf_block`` my original text had "function"  rather than "block", so how about:

```python
"""Load the next BGZF block of compressed data (PRIVATE)."""
```

I would suggest one pull request per file or related set of files (e.g. if you updated several entries under ``Bio/SeqIO/`` then one pull request makes sense). This is in part to help with reviewing as I won't always be the module owner.

**Comment by Gasta88** [comment]
Edit with the last comment.

**Comment by peterjc** [comment]
Thank you 👍

**Pull request merged**